packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Customization/src/CodeGenSerializationAttribute.cs

```diff
@@ -45,13 +45,13 @@ public CodeGenSerializationAttribute(string propertyName)
         public CodeGenSerializationAttribute(string propertyName, string serializationName)
         {
             PropertyName = propertyName;
-            SerializationPath = new[] { serializationName };
+            PropertySerializationName = serializationName;
         }
 
         public CodeGenSerializationAttribute(string propertyName, string[] serializationPath)
         {
             PropertyName = propertyName;
-            SerializationPath = serializationPath;
+            PropertySerializationName = serializationPath?[^1];
```

Comment:
If we don't think we need to keep this ctor, I can remove it.

Reply:
Yeah we should remove it if we are only allowing one value.